src/main/groovy/org/sharegov/mdcgis/PropertyInfoService.groovy

```diff
@@ -207,39 +214,42 @@ class PropertyInfoService {
 		if(!dataFromLayers)
 			return propertyType
 		
-		
-		if (dataFromLayers['MDC.PTXGIS']){
-			String propertyTypeFlag = 	dataFromLayers['MDC.PTXGIS'].attributes?.CONDO?:dataFromLayers['MDC.PTXGIS'].CONDO
-			switch (propertyTypeFlag){
-				case 'C':
-					propertyType = PropertyType.CONDO
-					break
-				case 'P':
-					propertyType = PropertyType.MULTI
-					break
-				case 'N':
-					propertyType = PropertyType.UNDEFINED
-					break
-				default:
-					propertyType = PropertyType.UNDEFINED
-
-			}
+		if (dataFromLayers['MDC.PaGIS']){
+			propertyType = getPropertyType(dataFromLayers, 'MDC.PaGIS')
 		}
-		else if (dataFromLayers['MDC.Parcels']) {
-			String propertyTypeFlag = 	dataFromLayers['MDC.Parcels'].attributes?.CONDO?:dataFromLayers['MDC.Parcels'].CONDO
-			switch (propertyTypeFlag){
-				case 'C':
-					propertyType = PropertyType.MULTI
-					break
-				default:
-					propertyType = PropertyType.UNDEFINED
-			}
+		else if (dataFromLayers['MDC.PaParcel']) {
+			propertyType = getPropertyType(dataFromLayers, 'MDC.PaParcel')
 		}
 
 		return propertyType
 	}
 
 	/**
+	 * Get Property type.
+	 *
+	 * @param dataFromLayers
+	 * @param layerName
+     * @return
+     */
+	private PropertyType getPropertyType(Map dataFromLayers, String layerName) {
+
+		PropertyType propertyType = PropertyType.UNDEFINED;
+		String isCondo = dataFromLayers[layerName].attributes?.CONDO_FLAG ?: dataFromLayers[layerName].CONDO_FLAG
+		String parentFolioNumber = dataFromLayers[layerName].attributes?.PARENT_FOLIO ?: dataFromLayers[layerName].PARENT_FOLIO
+		String folioNumber = dataFromLayers[layerName].attributes?.FOLIO ?: dataFromLayers[layerName].FOLIO
+		String referenceOnlyFlag = dataFromLayers[layerName].attributes?.REFERENCE_ONLY_FLAG ?: dataFromLayers[layerName].REFERENCE_ONLY_FLAG
+
+		if (isCondo.toBoolean()) {
+			propertyType = PropertyType.CONDO
+		} else if ( (parentFolioNumber != folioNumber && folioNumber.endsWith("0001")) && referenceOnlyFlag.toBoolean()) {
+			propertyType = PropertyType.MULTI
```

Comment:
if there are all && why do you need parentheses ((x && y) && zz) = (x && y && z)